src/main/java/org/opentripplanner/model/TripPattern.java

```diff
@@ -159,7 +159,11 @@ public void setHopGeometry(int i, LineString hopGeometry) {
      */
     public void setHopGeometriesFromPattern(TripPattern other) {
         this.hopGeometries = new byte[this.getStops().size() - 1][];
-        for (int i = 0; i < other.getStops().size() - 1; i++) {
+
+        // Gets the size of the shortest of the two trip patterns
+        int sizeOfShortestPattern = Math.min(this.getStops().size(), other.getStops().size());
```

Comment:
Should be sufficient to just add a comment here explaining why a length mismatch would occur (realtime update adding stops) and clarifying that this will only preserve geometry up to the added stop, at which point it will fall back on straight lines.

Reply:
I have added the comment now.